snapshotter/demux/metrics/proxy.go

```diff
@@ -30,10 +32,71 @@ const (
 	MaxMetricsResponseSize = 32768 // 32 KB
 )
 
+// Monitor is used for port and lifecycle management of metrics proxies.
+type Monitor struct {
+	// monitorChan indicates a port is free for metrics proxy server.
+	ch chan int
+	// portMap tracks used and unused metrics proxy server ports.
+	portMap map[int]bool
+	// startPort is first port in the configurable range of ports to serve metrics proxies from.
+	startPort int
+	// endPort is last port in the configurable range of ports to serve metrics proxies from.
+	endPort int
```

Comment:
Are these really needed after `portMap` allocation?

Reply:
They are needed for finding an open port within the port range with the current implementation of portMap as a set. I suppose we could truly treat portMap as a map, and initialize it on monitor creation to have all keys from `startPort` to `endPort` in the map populated with false values. With that approach, your suggested change of `range`ing over the map would fit. I'm not necessarily in favor of one over the over, do you have a preference or argument for one approach vs the other?

Reply:
Okay, I think understand your usage better now. That resolves my concern.